integration/aws.go

```diff
@@ -61,38 +50,77 @@ func NewAWSProvider(region string) (*AWSProvider, error) {
 }
 
 // DescribeAutoScalingGroups provides information about the available auto-scaling groups.
-func (p *AWSProvider) DescribeAutoScalingGroups() ([]AutoScalingGroup, error) {
-	fmt.Println("Retrieving data on autoscaling groups...")
+func (p *AWSProvider) DescribeAutoScalingGroups(names []string, scheme string, port int, path string) ([]AutoScalingGroup, error) {
+	fmt.Println("Retrieving data on autoscaling groups:", names)
+	start := time.Now()
 	svc := autoscaling.New(p.session)
 
-	groups, err := svc.DescribeAutoScalingGroups(nil)
+	awsGroups, err := svc.DescribeAutoScalingGroups(&autoscaling.DescribeAutoScalingGroupsInput{
+		AutoScalingGroupNames: convert(names),
+	})
 
 	if err != nil {
 		return nil, fmt.Errorf("failed to get the description of all autoscaling groups, %-v", err)
 	}
 
-	rv := make([]AutoScalingGroup, len(groups.AutoScalingGroups))
+	groups := make([]AutoScalingGroup, len(awsGroups.AutoScalingGroups))
+	errorCount := 0
+
+	for i, g := range awsGroups.AutoScalingGroups {
+		groupName := aws.StringValue(g.AutoScalingGroupName)
+		fmt.Printf("%s => Getting instance details for this autoscaling group.\n", groupName)
 
-	for i, g := range groups.AutoScalingGroups {
-		// Create the group.
-		asg := &AutoScalingGroup{
-			Name:      aws.StringValue(g.AutoScalingGroupName),
-			Instances: make([]Instance, len(g.Instances)),
+		instanceDetails, err := p.GetInstanceDetails(g.Instances, groupName, scheme, port, path)
+		if err != nil {
+			fmt.Printf("%s => Failed to get instance details, skipping this group\n", groupName)
+			errorCount++
+			continue
 		}
 
-		// Extract the instances and add them to the mapping.
-		for j, awsInstance := range g.Instances {
-			asg.Instances[j] = Instance{
-				ID:             aws.StringValue(awsInstance.InstanceId),
-				HealthStatus:   aws.StringValue(awsInstance.HealthStatus),
-				LifecycleState: aws.StringValue(awsInstance.LifecycleState),
-			}
+		asg := NewAutoScalingGroup(
+			aws.StringValue(g.AutoScalingGroupName),
+			g.Instances,
+			instanceDetails)
+
+		fmt.Printf("%s => Retrieved all instance details.", asg.Name)
+		groups[i] = asg
+	}
+
+	fmt.Println("time: *AWSProvider.DescribeAutoScalingGroups() ", time.Since(start))
+
+	if len(groups) == errorCount {
+		return nil, fmt.Errorf("No valid groups found.")
+	}
+
+	return groups, nil
+}
+
+func (p *AWSProvider) GetInstanceDetails(instances []*autoscaling.Instance, groupName string, scheme string, port int, path string) (InstanceDetails, error) {
+	start := time.Now()
+	details := InstanceDetails{}
+
+	for _, instance := range instances {
+		instanceId := aws.StringValue(instance.InstanceId)
```

Comment:
This should be called instanceID according to Go style guidelines. If you install the Go linter into your dev environment, it will alert you if you don't use Go conventions.